lib/src/test/java/io/ably/lib/test/rest/RestJWTTest.java

```diff
@@ -0,0 +1,157 @@
+package io.ably.lib.test.rest;
+
+import static org.junit.Assert.*;
+
+import io.ably.lib.test.common.Setup.Key;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.ably.lib.rest.AblyRest;
+import io.ably.lib.types.*;
+import io.ably.lib.rest.Auth.*;
+import io.ably.lib.test.common.ParameterizedTest;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class RestJWTTest extends ParameterizedTest {
+
+	private AblyRest restJWTRequester;
+	private ClientOptions jwtRequesterOptions;
+	private ClientOptions options;
+	private Key key = testVars.keys[0];
+	Param[] validKeys = new Param[]{ new Param("keyName", key.keyName), new Param("keySecret", key.keySecret) };
+	Param[] invalidKeys = new Param[]{ new Param("keyName", key.keyName), new Param("keySecret", "invalidinvalid") };
+	Param[] tokenEmbedded = new Param[]{ new Param("jwtType", "embedded") };
+	Param[] tokenEmbeddedAndEncrypted = new Param[]{ new Param("jwtType", "embedded"), new Param("encrypted", 1) };
+	Param[] jwtReturnType = new Param[]{ new Param("returnType", "jwt") };
+
+	@Before
+	public void setUpBefore() throws Exception {
+		jwtRequesterOptions = createOptions(testVars.keys[0].keyStr);
```

Comment:
Same comment as for the realtime tests. Please don't use the same options instances for every test - create new ones; if necessary add a helper method to do that.